src/aind_behavior_curriculum/tmp.py

```diff
@@ -0,0 +1,80 @@
+"""
+Work in Progress Curriculum
+"""
+
+from collections import defaultdict
+
+class Task:
+    """
+    Defined in exisiting PR
+    """
+    pass
+
+class Metric:
+    """
+    Initalizes a connection to relevant database (referencing some configuration file)
+    User extends this with their own metric implementation based on data inside the database.
+    """
+    pass
+
+
+class Evaluation:
+    """
+    Serializable Callable, takes in a list of metrics that is serialized alongside.
+    """
+    pass
+
+
+# TODO: De/serialization
+class Curriculum:
+    def __init__(self) -> None:
+        TaskId = int
+        # ID tasks on Task.name, a required field.
+        # Tasks are allowed to change parameters.
+        self.tasks: dict[str, TaskId] = {}
+        # Each node references a collection of (edge, destinatation node) associations.
+        # The collection is ordered by order of Curriculum API calls.
+        self.graph: dict[TaskId, dict[TaskId, Evaluation]] = defaultdict(dict)
```

Comment:
What is the advantage of having the nested dictionary? Shouldn't the value just be an array of Tuple(destination_stage, eval)? 
I can see a world where the keys will not be unique. For instance the same stage with multiple rules. Also, the order the this list matters so I think this is better suited to be an Array?